### PR TITLE
Solution to hamming excercise

### PR DESCRIPTION
**Pull request by tkerdonc**

### CODE REVIEW

File: go/hamming/hamming.go [hunk 1]
```diff
@@ -1,4 +1,37 @@
...
+)
+
+/*  Distance returns the hamming distance between two dna strands
+    passed as string parameters. Returns -1 if the strands are of
+	different lengths.
+    parameters :
+		a : strand as a string
+		b : strand as a string
+*/
+func Distance(a, b string) (distance int, distanceError error) {
```
Comment (ivajloip) [inline go/hamming/hamming.go]: while in some cases the named return parameters are useful, I think I have seen them being used only once _in the wild_.  The problem is that they bring a bit of confusion with them (where do the names come from and why there are no variables in the return statement). I am not sure I really like the use here, but I am also not strongly against it...

File: go/hamming/hamming.go [hunk 2]
```diff
@@ -1,4 +1,37 @@
...
+	if len(a) != len(b) {
+		distance = -1
```
Comment (ivajloip) [inline go/hamming/hamming.go]: do you need the `-1` value (I think the tests should be ok without it) given that you are returning an error? Generally if there is an error, all the other data is considered garbage and should not be relied upon. Therefore, the default value of 0 might be good enough...

File: go/hamming/hamming.go [hunk 3]
```diff
@@ -1,4 +1,37 @@
...
+		distanceError = errors.New("Distance expects strands to be of equal lengths.")
+	} else {
+		distanceError = nil
+		distance = 0
+		var aReader = strings.NewReader(a)
```
Comment (ivajloip) [inline go/hamming/hamming.go]: Given that the DNA is only composed of 4 different letters, I don't think Rune reading is really necessary. It doesn't hurt, but just taking the bytes of the strings (`a[i]` kind of thing) should be fine as well.

**Review by ivajloip — CHANGES_REQUESTED** [review]
As an idea, it's fine, but I have some recommendations :slightly_smiling_face:

File: go/hamming/hamming.go (hunk 1 above)
Comment (tkerdonc) [reply go/hamming/hamming.go]: No worries, I'll remove them (as well as in #8 and #7 ^^^)

File: go/hamming/hamming.go (hunk 3 above)
Comment (tkerdonc) [reply go/hamming/hamming.go]: Completely forgot we were dealing with DNA, silly me. applying your recommendations.
Am I correct in assuming that if the string could have contained more different characters than these 4 letters, some characters would not entirely be contained in the bytes returned by a single call to a[i], and therefore be counted twice or more ?

**Review by ivajloip — APPROVED** [review]
Seems good to me :slightly_smiling_face:

**Pull request merged**